src/fsharp/FSharp.DependencyManager/FSharp.DependencyManager.Utilities.fs

```diff
@@ -122,15 +122,38 @@ module Utilities =
         roots |> Array.tryFind(fun root -> msbuildPathExists root) |> msbuildOption
 #else
     let dotnetHostPath =
+        // How to find dotnet.exe --- woe is me; probing rules make me sad.
+        // Algorithm:
+        // 1. Look for DOTNET_HOST_PATH environment variable
+        //    this is the main user programable override .. provided by user to find a specific dotnet.exe
+        // 2. Probe for are we part of an .NetSDK install
+        //    In an sdk install we are always installed in:   sdk\3.0.100-rc2-014234\FSharp
+        //    dotnet or dotnet.exe will be found in the directory that contains the sdk directory
+        // 3. We are loaded in-process to some other application ... Eg. try .net
+        //    See if the host is dotnet.exe ... from netcoreapp3.0 on this is fairly unlikely
+        // 4. If it's none of the above we are going to have to rely on the path containing the way to find dotnet.exe
+        //
         if isRunningOnCoreClr then
             match (Environment.GetEnvironmentVariable("DOTNET_HOST_PATH")) with
-            | value when not (String.IsNullOrEmpty(value)) -> Some value                           // Value set externally
+            | value when not (String.IsNullOrEmpty(value)) ->
+                Some value                           // Value set externally
             | _ ->
-                let main = Process.GetCurrentProcess().MainModule
-                if main.ModuleName.StartsWith("dotnet") then
-                    Some main.FileName
+                // Probe for netsdk install
+                let dotnetLocation =
+                    let dotnetApp =
+                        let platform = Environment.OSVersion.Platform
+                        if platform = PlatformID.Unix then "dotnet" else "dotnet.exe"
+                    let assemblyLocation = typeof<DependencyManagerAttribute>.GetTypeInfo().Assembly.Location
+                    Path.Combine(assemblyLocation, "../../..", dotnetApp)
+
+                if File.Exists(dotnetLocation) then
+                    Some dotnetLocation
                 else
-                    None
+                    let main = Process.GetCurrentProcess().MainModule
+                    if main.ModuleName ="dotnet" then
+                        Some main.FileName
+                    else
+                        None
```

Comment:
Shouldn't this be `Some dotnetLocation` to fall back to the path?

Reply:
dotnetLocation is the full path to dotnet. or dotnet.exe.   dotnetApp is either dotnet. or dotnet.exe depending on OS.

This else codepath looks to see if it has been activated using dotnet.exe or dotnet, but the directory containing the dependency manager dll is not the FSharp directory, Ie, from a global tool or a unit test.  If so then the path to that dotnet.exe can be got from main.FileName.  The remaining case is the "we have no idea, lets just admit it" and so we return None.

Did that make sense?

Reply:
No, because in the case where we found this by using scripting from `dotnet-try`, the flow will be this:

```
if set %DOTNET_HOST_PATH% ... false
else
    dotnetLocation =
        assemblyLocation = DependencyManagerLocation = C:\Users\Brett\.dotnet\tools\.store\dotnet-try\1.0.44142.42\dotnet-try\1.0.44142.42\tools\netcoreapp3.0\any\FSharp.DependencyManager.dll
            => C:\Users\Brett\.dotnet\tools\.store\dotnet-try\1.0.44142.42\dotnet-try\1.0.44142.42\tools\netcoreapp3.0\any\..\..\..\dotnet.exe
            => C:\Users\Brett\.dotnet\tools\.store\dotnet-try\1.0.44142.42\dotnet-try\1.0.44142.42\dotnet.exe
    if exist dotnetLocation => false
    else
        main.ModuleName = "dotnet-try"
        => None

Finally, looking down to executeBuild, pathToExe is None so it just fails immediately.
```

Reply:
Hence the very last `None` should be `dotnet`/`dotnet.exe` to hope that it's on the path somewhere.

Reply:
@brettfo, yes, that makes sense, I typed rely on path in the algorithm, and then never dealt with it in code.  good eyes.